File: sdk/CsharpDotNet2/src/main/CsharpDotNet2/OrderCloud/Model/Product.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace OrderCloud.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class Product {
    /// <summary>
    /// Gets or Sets ID
    /// </summary>
    [DataMember(Name="ID", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ID")]
    public string ID { get; set; }

    /// <summary>
    /// Gets or Sets Name
    /// </summary>
    [DataMember(Name="Name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Name")]
    public string Name { get; set; }

    /// <summary>
    /// Gets or Sets Description
    /// </summary>
    [DataMember(Name="Description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Description")]
    public string Description { get; set; }

    /// <summary>
    /// Gets or Sets QuantityMultiplier
    /// </summary>
    [DataMember(Name="QuantityMultiplier", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "QuantityMultiplier")]
    public int? QuantityMultiplier { get; set; }

    /// <summary>
    /// Gets or Sets ShipWeight
    /// </summary>
    [DataMember(Name="ShipWeight", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ShipWeight")]
    public double? ShipWeight { get; set; }

    /// <summary>
    /// Gets or Sets Active
    /// </summary>
    [DataMember(Name="Active", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Active")]
    public bool? Active { get; set; }

    /// <summary>
    /// Gets or Sets Type
    /// </summary>
    [DataMember(Name="Type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Type")]
    public string Type { get; set; }

    /// <summary>
    /// Gets or Sets InventoryEnabled
    /// </summary>
    [DataMember(Name="InventoryEnabled", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "InventoryEnabled")]
    public bool? InventoryEnabled { get; set; }

    /// <summary>
    /// Gets or Sets InventoryNotificationPoint
    /// </summary>
    [DataMember(Name="InventoryNotificationPoint", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "InventoryNotificationPoint")]
    public int? InventoryNotificationPoint { get; set; }

    /// <summary>
    /// Gets or Sets VariantLevelInventory
    /// </summary>
    [DataMember(Name="VariantLevelInventory", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "VariantLevelInventory")]
    public bool? VariantLevelInventory { get; set; }

    /// <summary>
    /// Gets or Sets Xp
    /// </summary>
    [DataMember(Name="xp", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "xp")]
    public Object Xp { get; set; }

    /// <summary>
    /// Gets or Sets AllowOrderExceedInventory
    /// </summary>
    [DataMember(Name="AllowOrderExceedInventory", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "AllowOrderExceedInventory")]
    public bool? AllowOrderExceedInventory { get; set; }

    /// <summary>
    /// Gets or Sets InventoryVisible
    /// </summary>
    [DataMember(Name="InventoryVisible", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "InventoryVisible")]
    public bool? InventoryVisible { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Product {\n");
      sb.Append("  ID: ").Append(ID).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  Description: ").Append(Description).Append("\n");
      sb.Append("  QuantityMultiplier: ").Append(QuantityMultiplier).Append("\n");
      sb.Append("  ShipWeight: ").Append(ShipWeight).Append("\n");
      sb.Append("  Active: ").Append(Active).Append("\n");
      sb.Append("  Type: ").Append(Type).Append("\n");
      sb.Append("  InventoryEnabled: ").Append(InventoryEnabled).Append("\n");
      sb.Append("  InventoryNotificationPoint: ").Append(InventoryNotificationPoint).Append("\n");
      sb.Append("  VariantLevelInventory: ").Append(VariantLevelInventory).Append("\n");
      sb.Append("  Xp: ").Append(Xp).Append("\n");
      sb.Append("  AllowOrderExceedInventory: ").Append(AllowOrderExceedInventory).Append("\n");
      sb.Append("  InventoryVisible: ").Append(InventoryVisible).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: sdk/jaxrs-spec/src/main/java/io/swagger/model/Product.java
package io.swagger.model;

import java.math.BigDecimal;



import io.swagger.annotations.*;
import java.util.Objects;


public class Product   {
  
  private String ID = null;
  private String name = null;
  private String description = null;
  private Integer quantityMultiplier = null;
  private BigDecimal shipWeight = null;
  private Boolean active = null;
  private String type = null;
  private Boolean inventoryEnabled = null;
  private Integer inventoryNotificationPoint = null;
  private Boolean variantLevelInventory = null;
  private Object xp = null;
  private Boolean allowOrderExceedInventory = null;
  private Boolean inventoryVisible = null;

  /**
   **/
  public Product ID(String ID) {
    this.ID = ID;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public String getID() {
    return ID;
  }
  public void setID(String ID) {
    this.ID = ID;
  }

  /**
   **/
  public Product name(String name) {
    this.name = name;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  /**
   **/
  public Product description(String description) {
    this.description = description;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public String getDescription() {
    return description;
  }
  public void setDescription(String description) {
    this.description = description;
  }

  /**
   **/
  public Product quantityMultiplier(Integer quantityMultiplier) {
    this.quantityMultiplier = quantityMultiplier;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Integer getQuantityMultiplier() {
    return quantityMultiplier;
  }
  public void setQuantityMultiplier(Integer quantityMultiplier) {
    this.quantityMultiplier = quantityMultiplier;
  }

  /**
   **/
  public Product shipWeight(BigDecimal shipWeight) {
    this.shipWeight = shipWeight;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public BigDecimal getShipWeight() {
    return shipWeight;
  }
  public void setShipWeight(BigDecimal shipWeight) {
    this.shipWeight = shipWeight;
  }

  /**
   **/
  public Product active(Boolean active) {
    this.active = active;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Boolean getActive() {
    return active;
  }
  public void setActive(Boolean active) {
    this.active = active;
  }

  /**
   **/
  public Product type(String type) {
    this.type = type;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public String getType() {
    return type;
  }
  public void setType(String type) {
    this.type = type;
  }

  /**
   **/
  public Product inventoryEnabled(Boolean inventoryEnabled) {
    this.inventoryEnabled = inventoryEnabled;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Boolean getInventoryEnabled() {
    return inventoryEnabled;
  }
  public void setInventoryEnabled(Boolean inventoryEnabled) {
    this.inventoryEnabled = inventoryEnabled;
  }

  /**
   **/
  public Product inventoryNotificationPoint(Integer inventoryNotificationPoint) {
    this.inventoryNotificationPoint = inventoryNotificationPoint;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Integer getInventoryNotificationPoint() {
    return inventoryNotificationPoint;
  }
  public void setInventoryNotificationPoint(Integer inventoryNotificationPoint) {
    this.inventoryNotificationPoint = inventoryNotificationPoint;
  }

  /**
   **/
  public Product variantLevelInventory(Boolean variantLevelInventory) {
    this.variantLevelInventory = variantLevelInventory;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Boolean getVariantLevelInventory() {
    return variantLevelInventory;
  }
  public void setVariantLevelInventory(Boolean variantLevelInventory) {
    this.variantLevelInventory = variantLevelInventory;
  }

  /**
   **/
  public Product xp(Object xp) {
    this.xp = xp;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Object getXp() {
    return xp;
  }
  public void setXp(Object xp) {
    this.xp = xp;
  }

  /**
   **/
  public Product allowOrderExceedInventory(Boolean allowOrderExceedInventory) {
    this.allowOrderExceedInventory = allowOrderExceedInventory;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Boolean getAllowOrderExceedInventory() {
    return allowOrderExceedInventory;
  }
  public void setAllowOrderExceedInventory(Boolean allowOrderExceedInventory) {
    this.allowOrderExceedInventory = allowOrderExceedInventory;
  }

  /**
   **/
  public Product inventoryVisible(Boolean inventoryVisible) {
    this.inventoryVisible = inventoryVisible;
    return this;
  }

  
  @ApiModelProperty(example = "null", value = "")
  public Boolean getInventoryVisible() {
    return inventoryVisible;
  }
  public void setInventoryVisible(Boolean inventoryVisible) {
    this.inventoryVisible = inventoryVisible;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Product product = (Product) o;
    return Objects.equals(ID, product.ID) &&
        Objects.equals(name, product.name) &&
        Objects.equals(description, product.description) &&
        Objects.equals(quantityMultiplier, product.quantityMultiplier) &&
        Objects.equals(shipWeight, product.shipWeight) &&
        Objects.equals(active, product.active) &&
        Objects.equals(type, product.type) &&
        Objects.equals(inventoryEnabled, product.inventoryEnabled) &&
        Objects.equals(inventoryNotificationPoint, product.inventoryNotificationPoint) &&
        Objects.equals(variantLevelInventory, product.variantLevelInventory) &&
        Objects.equals(xp, product.xp) &&
        Objects.equals(allowOrderExceedInventory, product.allowOrderExceedInventory) &&
        Objects.equals(inventoryVisible, product.inventoryVisible);
  }

  @Override
  public int hashCode() {
    return Objects.hash(ID, name, description, quantityMultiplier, shipWeight, active, type, inventoryEnabled, inventoryNotificationPoint, variantLevelInventory, xp, allowOrderExceedInventory, inventoryVisible);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Product {\n");
    
    sb.append("    ID: ").append(toIndentedString(ID)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    quantityMultiplier: ").append(toIndentedString(quantityMultiplier)).append("\n");
    sb.append("    shipWeight: ").append(toIndentedString(shipWeight)).append("\n");
    sb.append("    active: ").append(toIndentedString(active)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    inventoryEnabled: ").append(toIndentedString(inventoryEnabled)).append("\n");
    sb.append("    inventoryNotificationPoint: ").append(toIndentedString(inventoryNotificationPoint)).append("\n");
    sb.append("    variantLevelInventory: ").append(toIndentedString(variantLevelInventory)).append("\n");
    sb.append("    xp: ").append(toIndentedString(xp)).append("\n");
    sb.append("    allowOrderExceedInventory: ").append(toIndentedString(allowOrderExceedInventory)).append("\n");
    sb.append("    inventoryVisible: ").append(toIndentedString(inventoryVisible)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
